sdk/servicebus/azure-servicebus/tests/async_tests/test_sb_client_async.py

```diff
@@ -443,3 +443,113 @@ async def test_backoff_fixed_retry(self):
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    async def test_default_client_id_queue_sender(self, servicebus_namespace_connection_string):
+        servicebus_connection_str = "connection_string"
+        queue_name = "queue_name"
+        servicebus_client = ServiceBusClient.from_connection_string(conn_str=servicebus_connection_str)
+        async with servicebus_client:
+            queue_sender = servicebus_client.get_queue_sender(queue_name=queue_name)
+            assert queue_sender.identifier is not None
+            assert "SBSender" in queue_sender.identifier
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    async def test_custom_client_id_queue_receiver(self, servicebus_namespace_connection_string):
+        servicebus_connection_str = "connection_string"
+        queue_name = "queue_name"
+        custom_id = "my_custom_id"
+        servicebus_client = ServiceBusClient.from_connection_string(conn_str=servicebus_connection_str)
+        async with servicebus_client:
+            queue_receiver = servicebus_client.get_queue_receiver(queue_name=queue_name, client_identifier=custom_id)
+            assert queue_receiver.identifier is not None
+            assert queue_receiver.identifier == custom_id
+
+    
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    async def test_default_client_id_queue_receiver(self, servicebus_namespace_connection_string):
+        servicebus_connection_str = "connection_string"
+        queue_name = "queue_name"
+        servicebus_client = ServiceBusClient.from_connection_string(conn_str=servicebus_connection_str)
+        async with servicebus_client:
+            queue_receiver = servicebus_client.get_queue_receiver(queue_name=queue_name)
+            assert queue_receiver.identifier is not None
+            assert "SBReceiver" in queue_receiver.identifier
+
+    @pytest.mark.liveTest
+    @pytest.mark.live_test_only
+    @CachedResourceGroupPreparer(name_prefix='servicebustest')
+    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
+    async def test_custom_client_id_topic_sender(self, servicebus_namespace_connection_string):
+        servicebus_connection_str = "connection_string"
+        queue_name = "queue_name"
+        custom_id = "my_custom_id"
+        servicebus_client = ServiceBusClient.from_connection_string(conn_str=servicebus_connection_str)
+        async with servicebus_client:
+            topic_sender = servicebus_client.get_queue_receiver(queue_name=queue_name, client_identifier=custom_id)
+            assert topic_sender.identifier is not None
+            assert topic_sender.identifier == custom_id
```

Comment:
think you'll need to get the cached topic, to ensure that it exists:
```suggestion
    @CachedResourceGroupPreparer(name_prefix='servicebustest')
    @CachedServiceBusNamespacePreparer(name_prefix='servicebustest')
    @CachedServiceBusTopicPreparer(name_prefix='servicebustest')
    async def test_custom_client_id_topic_sender(self, servicebus_namespace_connection_string, servicebus_topic):
        custom_id = "my_custom_id"
        servicebus_client = ServiceBusClient.from_connection_string(servicebus_namespace_connection_string)
        async with servicebus_client:
            topic_sender = servicebus_client.get_topic_sender(servicebus_topic.name, client_identifier=custom_id)
            assert topic_sender.identifier is not None
            assert topic_sender.identifier == custom_id
```